great_expectations/experimental/datasources/file_path_data_asset.py

```diff
@@ -65,7 +71,8 @@ class _FilePathDataAsset(DataAsset):
     _all_group_index_to_group_name_mapping: Dict[int, str] = pydantic.PrivateAttr()
     _all_group_names: List[str] = pydantic.PrivateAttr()
 
-    _data_connector: DataConnector | None = pydantic.PrivateAttr()
+    _data_connector: DataConnector = pydantic.PrivateAttr()
```

Comment:
We redeclare `_data_connector` here, because in `_DataAsset`, it is optional by design (but here it is required, because all File-Path DataAsset implementations rely on a `DataConnector`).